standard-libraries/tests/list-tests.3.mr

```diff
@@ -0,0 +1,86 @@
+## Tests for generic lists
+
+test list-append-test()
+  var List{String} list
+  list.append(owner "first")
+  list.append(owner "second")
+  list.append(owner "third")
+  var ListIterator{String} iter
+  iter.init(user list)
```

Comment:
Moreover, I'm thinking to add an `Iterable` interface.
The current problem with this is how to handle memory management, because I don't want to force a dynamic allocation of the iterator.